crates/connectors/ndc-postgres/src/schema.rs

```diff
@@ -252,16 +252,18 @@ pub async fn get_schema(
         })
         .collect();
 
+    let mut more_object_types = BTreeMap::new();
     let generated_procedures: Vec<models::ProcedureInfo> =
         query_engine_translation::translation::mutation::generate::generate(
             &metadata.tables,
             &config.mutations_version,
         )
         .iter()
-        .map(|(name, mutation)| mutation_to_procedure(name, mutation))
+        .map(|(name, mutation)| mutation_to_procedure(name, mutation, &mut more_object_types))
```

Comment:
I am really not a fan of a `map` which mutates. I suggest you either return procedures and types, aggregating them in `unzip`.

Nothing reads from the value, as far as I can tell, so this also affords way more control than is necessary.

Reply:
> I am really not a fan of a map which mutates.

Why not?

> I suggest you either return procedures and types, aggregating them in unzip.

I actually tried it first and it was very clumsy, which is why I moved to mutating.

Reply:
Yeah, I just tried it too and you're right, it's a bit too clunky.

Makes me sad, but 🤷🏾

Reply:
Yeah I'd probably avoid mutation if this was nicer, but personally I feel like using mutation when it's localized is absolutely fine.

Reply:
I will just note that this has been solved above by iterating twice.

Not sure if that's better or worse.